File: Plugins/VanillaBasics/src/main/java/org/tobi29/scapes/vanilla/basics/packet/PacketNotification.java
/*
 * Copyright 2012-2015 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.tobi29.scapes.vanilla.basics.packet;

import org.tobi29.scapes.chunk.WorldClient;
import org.tobi29.scapes.client.connection.ClientConnection;
import org.tobi29.scapes.client.gui.desktop.GuiInGameMessage;
import org.tobi29.scapes.engine.utils.io.ReadableByteStream;
import org.tobi29.scapes.engine.utils.io.WritableByteStream;
import org.tobi29.scapes.packets.Packet;
import org.tobi29.scapes.packets.PacketClient;
import org.tobi29.scapes.server.connection.PlayerConnection;

import java.io.IOException;

public class PacketNotification extends Packet implements PacketClient {
    private String title, text;

    public PacketNotification() {
    }

    public PacketNotification(String title, String text) {
        this.title = title;
        this.text = text;
    }

    @Override
    public void sendClient(PlayerConnection player, WritableByteStream stream)
            throws IOException {
        stream.putString(title);
        stream.putString(text);
    }

    @Override
    public void parseClient(ClientConnection client, ReadableByteStream stream)
            throws IOException {
        title = stream.getString();
        text = stream.getString();
    }

    @Override
    public void runClient(ClientConnection client, WorldClient world) {
        client.entity().openGui(new GuiInGameMessage(client.game(), title, text,
                client.game().engine().guiStyle()));
    }
}

File: Plugins/VanillaBasics/src/main/kotlin/scapes/plugin/tobi29/vanilla/basics/packet/PacketNotification.kt
/*
 * Copyright 2012-2016 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package scapes.plugin.tobi29.vanilla.basics.packet

import org.tobi29.scapes.client.connection.ClientConnection
import org.tobi29.scapes.client.gui.GuiInGameMessage
import org.tobi29.scapes.engine.utils.io.ReadableByteStream
import org.tobi29.scapes.engine.utils.io.WritableByteStream
import org.tobi29.scapes.packets.PacketAbstract
import org.tobi29.scapes.packets.PacketClient
import org.tobi29.scapes.server.connection.PlayerConnection

class PacketNotification : PacketAbstract, PacketClient {
    private lateinit var title: String
    private lateinit var text: String

    constructor()

    constructor(title: String, text: String) {
        this.title = title
        this.text = text
    }

    override fun sendClient(player: PlayerConnection,
                            stream: WritableByteStream) {
        stream.putString(title)
        stream.putString(text)
    }

    override fun parseClient(client: ClientConnection,
                             stream: ReadableByteStream) {
        title = stream.string
        text = stream.string
    }

    override fun runClient(client: ClientConnection) {
        client.mob {
            it.openGui(GuiInGameMessage(client.game, title, text,
                    client.game.engine.guiStyle))
        }
    }
}
